src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Appearance/ApplicationThemeManager.cs

```diff
@@ -0,0 +1,278 @@
+// This Source Code Form is subject to the terms of the MIT License.
+// If a copy of the MIT was not distributed with this file, You can obtain one at https://opensource.org/licenses/MIT.
+// Copyright (C) Leszek Pomianowski and WPF UI Contributors.
+// All Rights Reserved.
+
+using System.Runtime.CompilerServices;
+using System.Windows.Controls;
+using System.Windows;
+
+namespace System.Windows.Appearance;
+
+/// <summary>
+/// Allows to manage the application theme by swapping resource dictionaries containing dynamic resources with color information.
+/// </summary>
+/// <example>
+/// <code lang="csharp">
+/// ApplicationThemeManager.Apply(
+///     ApplicationTheme.Light
+/// );
+/// </code>
+/// <code lang="csharp">
+/// if (ApplicationThemeManager.GetAppTheme() == ApplicationTheme.Dark)
+/// {
+///     ApplicationThemeManager.Apply(
+///         ApplicationTheme.Light
+///     );
+/// }
+/// </code>
+/// <code>
+/// ApplicationThemeManager.Changed += (theme, accent) =>
+/// {
+///     Debug.WriteLine($"Application theme changed to {theme.ToString()}");
+/// };
+/// </code>
+/// </example>
+public static class ApplicationThemeManager
+{
+    private static ApplicationTheme _cachedApplicationTheme = ApplicationTheme.Unknown;
+
+    internal const string LibraryNamespace = "win11;";
+
+    internal const string ThemesDictionaryPath = "pack://application:,,,/PresentationFramework.Win11;component/Resources/Theme/";
+
+    /// <summary>
+    /// Event triggered when the application's theme is changed.
+    /// </summary>
+    public static event ThemeChangedEvent Changed;
```

Comment:
Should we use nullable?

```suggestion
    public static event ThemeChangedEvent? Changed;
```

Reply:
Yes. Had to remove it in order to fix a build issue that complained about use of nullability in nullable contexts.
I've enabled nullability for this file currently.